### PR TITLE
#61 - option to create password protected PDFs

### PR DESCRIPTION
**Pull request by anandkumarkparmar**

### CODE REVIEW

**Comment by anandkumarkparmar** [comment]
Please review the PR @Swati4star

File: app/build.gradle [hunk 1]
```diff
@@ -48,4 +48,6 @@ dependencies {
     implementation 'com.jakewharton:butterknife:8.5.1'
     implementation 'id.zelory:compressor:2.1.0'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
+
```
Comment (Swati4star) [inline app/build.gradle]: Remove this extra line

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 2]
```diff
@@ -428,6 +415,15 @@ protected String doInBackground(String... params) {
 
                 Log.v("Stage 3", "Pdf writer");
 
+                if (mPassword != null && !mPassword.trim().equals("")) {
+                    writer.setEncryption(mPassword.getBytes(),
+                            getString(R.string.app_name).getBytes(),
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: Is there any reason for setting app name as the owner password?

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 3]
```diff
@@ -528,4 +525,46 @@ private void resetValues() {
...
+            return;
+        }
+
+        new MaterialDialog.Builder(mActivity)
+                .title(R.string.set_password)
+                .content(R.string.enter_password)
+                .input(getString(R.string.example_password), null, new MaterialDialog.InputCallback() {
+                    @Override
+                    public void onInput(@NonNull MaterialDialog dialog, CharSequence input) {
+                        if (input == null || input.toString().trim().equals("")) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: `input == null || input.toString().trim().equals("")`
This check is being used many times. Move this validation to a new class (`StringUtils`) in [util](https://github.com/Swati4star/Images-to-PDF/tree/master/app/src/main/java/swati4star/createpdf/util).

**Comment by Swati4star** [comment]
Suppose I have added the password to the PDF. Now, I want to remove it. Is there any option to do so right now?

Just giving a suggestion : 
- When I click on this `password protect PDF` for the first time, the behaviour should be as implemented.
- You can add a big tick over the `password protect PDF` if there is some password set.
- When I click on this `password protect PDF` option now, the new dialog should have an option to remove the password also. Also, autofill the previous password.


Another UI enhancement suggestion(Not compulsory to implement) :
screen shot 2018-06-14 at 12 20 42 am
Add an eye by the side of passowrd field. So, the user can hide/view the password

**Comment by Swati4star** [comment]
@anandkumarkparmar you are working on this?

**Comment by anandkparmar** [comment]
@Swati4star Yes I am working on this.

**Comment by anandkumarkparmar** [comment]
@Swati4star Yes.. will push the code soon

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java (hunk 2 above)
Comment (anandkumarkparmar) [reply app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: No. Can you suggest any other option, which we can set as the owner password? @Swati4star

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: No problem. Let it be the owner password only.

**Comment by anandkumarkparmar** [comment]
Hi @Swati4star, I haven't added hide/view password functionality. I guess, with current _MaterialDialog_, we can not do that. So, we have to go with Custom Dialog and for that we can create a different PR.

**Comment by Swati4star** [comment]
The 'Update feedback method' commit (72c759a23ec75e232d92555fdf28b73ca4a46d83P) is also here in your PR. Look into that.

File: app/build.gradle [hunk 4]
```diff
@@ -48,4 +48,5 @@ dependencies {
     implementation 'com.jakewharton:butterknife:8.5.1'
     implementation 'id.zelory:compressor:2.1.0'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
+    implementation 'com.madgag.spongycastle:core:1.58.0.0'
```
Comment (Swati4star) [inline app/build.gradle]: Why are you adding this?

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 5]
```diff
@@ -94,11 +94,11 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
     private void getFeedback() {
-        Intent intent = new Intent(Intent.ACTION_SEND);
-        intent.setType("message/rfc822");
-        intent.putExtra(Intent.EXTRA_EMAIL  , new String[]{"swari4star@gmail.com"});
+        Intent intent = new Intent(Intent.ACTION_SENDTO);
+        intent.setData(Uri.parse("mailto:")); // only email apps should handle this
+        intent.putExtra(Intent.EXTRA_EMAIL, new String[]{"swati4star@gmail.com"});
         intent.putExtra(Intent.EXTRA_SUBJECT, getResources().getString(R.string.feedback_subject));
-        intent.putExtra(Intent.EXTRA_TEXT   , getResources().getString(R.string.feedback_text));
+        intent.putExtra(Intent.EXTRA_TEXT, getResources().getString(R.string.feedback_text));
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: These changes shouldn't be a part of this feature.

File: app/build.gradle (hunk 4 above)
Comment (anandkumarkparmar) [reply app/build.gradle]: iText is dependent on spongycastle library for encryption of password.

Please refer this: https://viralpatel.net/blogs/password-protect-pdf-itext-java/

Comment (Swati4star) [reply app/build.gradle]: okay. cool 👍

**Comment by Swati4star** [comment]
Merged! Thanks 👍

**Pull request merged**